src/content/app/genome-browser/hooks/useBrowserRouting.ts

```diff
@@ -171,6 +173,17 @@ const useBrowserRouting = () => {
     }
   }, [genomeId]);
 
+  useEffect(() => {
+    if (genomeId && trackCategoriesForGenome) {
+      dispatch(
+        setInitialTrackConfigsPerGenome({
+          genomeId,
+          trackCategories: trackCategoriesForGenome
+        })
+      );
+    }
+  }, [genomeId, trackCategoriesForGenome]);
```

Comment:
We shouldn't be doing this here. It doesn't have anything to do with the routing. If we want to restore the data saved to the browser storage, we should probably run this in the Root component (see `useRestoredReduxState` hook)

Reply:
✅